File: Gravitryp v2/src/objects/Planet.java
package objects;

public class Planet extends Sprite{

	private double gc; // gravity constant
	
	public double getGc() {
		return gc;
	}

	public void setGc(double gc) {
		this.gc = gc;
	}

	public Planet(Point centre, double radius, double gc) {
		super(centre, Sprite.CIRCLE, 2*radius, 2*radius, new Point(1, 0));
		this.setGc(gc);
	}
}

File: OnlineGravitryp/WebContent/resources/Planet.js
/**
 * 
 */
function Planet(centre, radius, gc) {
	
	this.sprite = new Sprite(centre, "circle", 2*radius, 2*radius, new Point(1, 0));
	this.radius = radius;
	this.gc = gc;
	
	this.relocate = function(newx, newy) {
		this.sprite.centre = new Point(newx, newy);
		this.sprite.setHitbox(this.sprite.centre, new Point(1, 0));
	}
}